src/formatting-model-api.d.ts

```diff
@@ -0,0 +1,582 @@
+/**
+ * Formatting Pane Model Interfaces
+ */
+
+declare module powerbi {
+    export namespace visuals {
+        export type FormattingModelWarnings = { [cardUid: string]: IVisualWarning };
+
+        export interface FormattingModel {
+            cards: (FormattingCard | FormattingCardPlaceholder)[];
+            modelUnavailableImageName?: string;
+            modelUnavailableMessageKey?: string;
+            modelUnavailableLearnMoreLink?: string;
+            // this is used for displaying messages in formatting pane, with an action to usually open some dialog.
+            formattingModelMessage?: FormattingModelMessage;
+            warnings?: FormattingModelWarnings;
+        }
+
+        export interface FormattingModelMessage {
+            messageKey: string;
+            action?: FormattingModelMessageAction;
+        }
+
+        export interface FormattingModelMessageAction {
+            key: string;
+            fn: () => void;
+        }
+
+        export interface FormattingWarning {
+            warningMessage?: IVisualErrorMessage;
+        }
+
+        export interface FormattingCard extends VisualFormattingUid, VisualFormattingDisplayName, FormattingWarning {
+            disabled?: boolean;
+            /** card disabled reason */
+            disabledReason?: string;
+            /** This is top-level card On/Off property */
+            topLevelToggle?: EnabledSlice;
+            groups: (FormattingGroup | FormattingGroupPlaceholder)[];
+            /** This contains the list of descriptors for revert to default, note the descriptors of all the slices ever possible in the card must be added */
+            revertToDefaultDescriptors?: FormattingDescriptor[];
```

Comment:
hey can you explain what the purpose of this is?

Reply:
revertToDefaultDescriptors is the list of all formatting properties descriptors (descriptor is objectName and propertyName - the same as in capabilites) in this formatting card that will be reverted/reset their values to their default values when we click on `Reset to default` option which appears in each formatting card.
revertToDefaultDescriptors is required if we want to use the `Reset to default` option and reset all of the properties in card in one click